lib/pages/search/search_view.dart

```diff
@@ -186,22 +251,140 @@ class SearchView extends StatelessWidget {
         mainAxisAlignment: MainAxisAlignment.start,
         children: [
           Expanded(
-            child: ValueListenableBuilder(
-              valueListenable: searchController.textEditingController,
-              builder: (context, value, child) {
-                return Text(
-                  value.text.isEmpty
-                      ? L10n.of(context)!.recent
-                      : L10n.of(context)!.chatsAndContacts,
-                  style: Theme.of(context).textTheme.labelLarge?.copyWith(
-                        color: LinagoraRefColors.material().neutral[40],
-                      ),
-                );
-              },
+            child: Text(
+              headerText,
+              style: Theme.of(context).textTheme.labelLarge?.copyWith(
+                    color: LinagoraRefColors.material().neutral[40],
+                  ),
             ),
           ),
+          if (needShowMore)
+            InkWell(
+              onTap: () {
+                searchController.searchContactAndRecentChatController!
+                    .toggleShowMore();
+              },
+              child: ValueListenableBuilder(
+                valueListenable: searchController
+                    .searchContactAndRecentChatController!
+                    .isShowChatsAndContactsNotifier,
+                builder: (context, isShowMore, child) {
+                  if (isShowMore) {
+                    return Text(
+                      L10n.of(context)!.showMore,
+                      style: Theme.of(context).textTheme.labelLarge?.copyWith(
+                            color: LinagoraRefColors.material().neutral[40],
+                          ),
+                    );
+                  } else {
+                    return Text(
+                      L10n.of(context)!.showLess,
+                      style: Theme.of(context).textTheme.labelLarge?.copyWith(
+                            color: LinagoraRefColors.material().neutral[40],
+                          ),
+                    );
```

Comment:
same styles so just do that imo 
```suggestion
                    return Text(
                      isShowMore ? L10n.of(context)!.showMore : L10n.of(context)!.showLess,
                      style: Theme.of(context).textTheme.labelLarge?.copyWith(
                            color: LinagoraRefColors.material().neutral[40],
                          ),
                    );
```